index.js

```diff
@@ -236,41 +254,43 @@ const image = {
   },
   getOriginal (req, res) {
     const matches = parsing.getImageParams(req);
-    log.log('info', "Requested original image " + matches.fileName + " in format " + matches.fileType);
-    if (parsing.supportedFileType(matches.fileType)) {
-      const file = `${config.get('originals_dir')}/${matches.fileName}`;
-      fs.exists(file, function (exists) {
-        if (!exists) {
-          return helpers.send404(res, file);
-        }
-        let headers = {
-          'Content-Type': parsing.supportedFileType(matches.fileType),
-          'Cache-Control': 'public',
-          'Etag': `${matches.fileName}_${matches.fileType}`,
-          'Expires': helpers.farFutureDate()
-        };
-        res.writeHead(200, headers);
-        fs.readFile(file, function (err, data) {
-          res.end(data);
-        });
-      });
+    log.log('info', `Requested original image ${matches.fileName} in format ${matches.fileType}`);
+    if (!parsing.supportedFileType(matches.fileType)) {
+      return helpers.set404(res, req.url);
     }
+    const file = `${config.get('originals_dir')}/${matches.fileName}`;
+    fs.exists(file, (exists) => {
+      if (!exists) {
+        log.log('warn', 'Image ${matches.fileName} is not available locally');
+        return helpers.send404(res, file);
+      }
+      let headers = {
+        'Content-Type': parsing.supportedFileType(matches.fileType),
+        'Cache-Control': 'public',
+        'Etag': `${matches.fileName}_${matches.fileType}`,
+        'Expires': helpers.farFutureDate()
+      };
+      res.writeHead(200, headers);
+      fs.readFile(file, (err, data) => {
+        res.end(data);
+      });
+    });
   }
 };
 
 function correctlyResize(file, params, callback) {
```

Comment:
It used to be done at several places but as you can see in the logic it really isnt that straight forward anymore